atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraTracingConfig.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableCassandraTracingConfig.class)
+@JsonSerialize(as = ImmutableCassandraTracingConfig.class)
+public abstract class CassandraTracingConfig {
+    private static final Logger log = LoggerFactory.getLogger(CassandraTracingConfig.class);
+    private static final AtomicBoolean loggedWarning = new AtomicBoolean();
+
+    @JsonProperty("enabled")
+    @Value.Default
+    public boolean enabled() {
+        return false;
+    }
+
+    @JsonProperty("trace-probability")
+    @Value.Default
+    public double traceProbability() {
+        return 1.0;
+    }
+
+    @JsonProperty("tables-to-trace")
+    public abstract Set<String> tablesToTrace();
+
+    @JsonProperty("min-duration-to-log-ms")
+    @Value.Default
+    public int minDurationToLogMs() {
+        return 0;
+    }
+
+    public boolean shouldTraceQuery(String tablename, Random random) {
+        if (!enabled()) {
+            return false;
+        }
+        if (tablesToTrace().contains(tablename)) {
+            if (traceProbability() == 1.0) {
+                return true;
+            } else {
+                if (random.nextDouble() <= traceProbability()) {
+                    return true;
+                }
+            }
+        }
+        if (tablesToTrace().isEmpty()) {
+            return true; // accept enabled = true but no tables specified to mean trace all tables
+        }
```

Comment:
I know that this was the previous behaviour, but it's really... weird. If my read is correct, it's basically saying that if you leave the tables to trace as empty, you also want to ignore the tracing probability, which I'm fairly certain is not what was intended. It seems to me that the above contains check should probably have been:

```java
if (tablesToTrace().isEmpty() || tablesToTrace().contains(tablename)) {
```

which makes more sense. And given that users will be forced to change their config to their Cassandra KVS config anyway, I'd push in that direction (so long as this is clearly reflected in the docs).

Reply:
can we make changes to behavior in a separate change?

Reply:
Done